src/azul/service/repository_controller.py

```diff
@@ -35,11 +39,66 @@ def service(self):
     def repository_plugin(cls, catalog: CatalogName) -> RepositoryPlugin:
         return RepositoryPlugin.load(catalog).create(catalog)
 
+    @classmethod
+    def parse_range_request_header(cls,
+                                   headers: Mapping[str, str]
+                                   ) -> Sequence[Tuple[Optional[int], Optional[int]]]:
+        """
+        >>> f = RepositoryController.parse_range_request_header
+        >>> f({})
+        []
+
+        >>> f({'range': 'bytes=100-200,300-400'})
+        [(100, 200), (300, 400)]
+
+        >>> f({'range': 'bytes=-100'})
+        [(None, 100)]
+
+        >>> f({'range': 'bytes=100-'})
+        [(100, None)]
+
+        >>> f({'range': ''})
+        []
+
+        >>> f({'range': 'foo=100'})
+        []
+
+        >>> f({'range': 'bytes=100-200-'})
+        Traceback (most recent call last):
+        ...
+        chalice.app.BadRequestError: BadRequestError: Invalid range specifier 'bytes=100-200-'
+        """
+
+        def to_int_or_none(value: str) -> Optional[int]:
+            try:
+                value = int(value)
+            except ValueError:
+                assert value == '', value
```

Comment:
Now you're going the other extreme. We use EAFP for efficiency and simplicity. Here EAFP makes the code really complicated. Why not just:

```python
def to_int_or_none(s):
    return None if s == '' else int(s)
to_int_or_none('')
to_int_or_none('x')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<input>", line 2, in to_int_or_none
ValueError: invalid literal for int() with base 10: 'x'
to_int_or_none('1')
1
```